File: android-java/app/src/main/java/com/oursky/skeleton/helper/Touchable.java
package com.oursky.skeleton.helper;

import android.support.annotation.NonNull;
import android.view.MotionEvent;
import android.view.View;

@SuppressWarnings({"WeakerAccess", "unused"})
public class Touchable {
    public static void make(@NonNull View view) {
        view.setOnTouchListener((v,ev) -> {
            switch(ev.getAction()) {
                case MotionEvent.ACTION_DOWN:
                    if (!v.isEnabled()) break;
                    v.animate().cancel();
                    v.animate().alpha(0.2f).setDuration(150);
                    return true;
                case MotionEvent.ACTION_CANCEL:
                    if (!v.isEnabled()) break;
                    v.animate().cancel();
                    v.animate().alpha(1.0f).setDuration(150);
                    break;
                case MotionEvent.ACTION_UP:
                    if (!v.isEnabled()) break;
                    v.animate().cancel();
                    v.animate().alpha(1.0f).setDuration(150);
                    v.performClick();
                    break;
            } return false;
        });
    }
}

File: android-kotlin/app/src/main/java/com/oursky/skeleton/helper/Touchable.kt
package com.oursky.skeleton.helper

import android.view.MotionEvent
import android.view.View

@Suppress("unused")
object Touchable {
    fun make(view: View) {
        view.setOnTouchListener { v, ev ->
            when (ev.action) {
                MotionEvent.ACTION_DOWN -> {
                    if (v.isEnabled) {
                        v.animate().cancel()
                        v.animate().alpha(0.2f).duration = 150
                        return@setOnTouchListener true
                    }
                }
                MotionEvent.ACTION_CANCEL -> {
                    if (v.isEnabled) {
                        v.animate().cancel()
                        v.animate().alpha(1.0f).duration = 150
                    }
                }
                MotionEvent.ACTION_UP -> {
                    if (v.isEnabled) {
                        v.animate().cancel()
                        v.animate().alpha(1.0f).duration = 150
                        v.performClick()
                    }
                }
            }
            false
        }
    }
}
